src/routes/categories/containers/CategoriesContainer.js

```diff
@@ -56,11 +52,20 @@ const mapStateToProps = (state: StateType, ownProps) => {
   }
 }
 
+const mapDispatchToProps = (dispatch: Dispatch<StoreActionType>, ownProps: OwnPropsType) => ({
+  navigateToCategory: createNavigateToCategory('Categories', dispatch, ownProps.navigation),
+  navigateToIntegreatUrl: createNavigateToIntegreatUrl(dispatch, ownProps.navigation),
+  changeUnavailableLanguage: (city: string, newLanguage: string) => dispatch({
+    type: 'SWITCH_CONTENT_LANGUAGE',
+    params: {city, newLanguage}
+  })
+})
+
 export default compose([
-  withRouteCleaner,
   connect(mapStateToProps, mapDispatchToProps),
-  // TODO NATIVE-112 Show errors
-  branch(props => props.error, renderComponent(Failure)),
-  branch(props => props.invalidLanguage, renderComponent(CategoryLanguageNotAvailableContainer)),
+  withRouteCleaner,
+  withError,
+  withLanguageNotAvailable,
+  withLoading,
```

Comment:
I'm not sure whether this is getting a bit out of hands with hocs. I don't have a better idea right now.
Maybe we can make withError and withLoading a single HOC.

Reply:
Yep, we could just use a single HOC for loading, errors and language not available, which is also kind of an error.